qutebrowser/browser/webkit/downloads.py

```diff
@@ -816,27 +836,10 @@ def get_request(self, request, *, fileobj=None, filename=None,
         if suggested_fn is None:
             suggested_fn = 'qutebrowser-download'
 
-        # We won't need a question if a filename or fileobj is already given
-        if fileobj is None and filename is None:
-            filename, q = ask_for_filename(
-                suggested_fn, self._win_id, parent=self,
-                prompt_download_directory=prompt_download_directory
-            )
-
-        if fileobj is not None or filename is not None:
-            return self.fetch_request(request,
-                                      fileobj=fileobj,
-                                      filename=filename,
-                                      suggested_filename=suggested_fn,
-                                      **kwargs)
-        q.answered.connect(
-            lambda fn: self.fetch_request(request,
-                                          filename=fn,
-                                          suggested_filename=suggested_fn,
-                                          **kwargs))
-        self._postprocess_question(q)
-        q.ask()
-        return None
+        return self.fetch_request(request,
+                                  target=target,
+                                  suggested_filename=suggested_fn,
+                                  **kwargs)
```

Comment:
> First, the old code had filename questions in DownloadManager.get and
> DownloadManager.fetch which were almost identical, thus the part in
> DownloadManager.get was removed in an earlier commit. All filename
> asking is now done by DownloadManager.fetch. The good part is code
> deduplication, the bad part is slightly modified behavior: The new code
> doesn't wait for a filename to start the download, instead it tries to
> fill the buffer immediately. This made the test fail because qute:// has
> no registered handler, so in order for the test to pass now, the "no
> crash" part is not enough, we also need to expect the "No handler"
> error.

So the reason why the download isn't started immediately here is that it's
quite problematic to have a download without downloading anything - so I tried
to avoid that if possible.

However, nowadays I got this mystery solved pretty well I think (by doing the
"slowly download to RAM" thing), so I'm fine with starting the download
immediately here as it'll simplify stuff quite a bit.